### PR TITLE
Feature/cc 1229

### PR DESCRIPTION
**Pull request by dtoulcher-ch**
__Short description outlining key changes/additions__

__JIRA Ticket Number__

### Type of change

* [ ] Bug fix
* [ ] New feature
* [ ] Breaking change
* [ ] Infrastructure change

### Pull request checklist

* [ ] I have added unit tests for new code that I have added
* [ ] I have added/updated functional tests where appropriate
* [ ] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/accounts/user/configuration/InterceptorConfig.java [hunk 1]
```diff
@@ -45,22 +45,20 @@ public void addInterceptors(@NonNull final InterceptorRegistry registry) {
...
+        registry.addInterceptor(loggingInterceptor)
+        .addPathPatterns(WILDCARD);
     }
      
     private void addInternlUserAdminSearchInterceptor( final InterceptorRegistry registry){
         registry.addInterceptor(new RolePermissionInterceptor(AccountsUserServiceApplication.applicationNameSpace, "/admin/search"))
-            .addPathPatterns(WILDCARD)
-            .excludePathPatterns(HEALTH_ENDPOINT, USERS_ENDPOINTS)
-            .order(1);
+        .addPathPatterns(INTERNAL_USERS_ENDPOINTS, ADMIN_ROLE_ENDPOINTS);
```
Comment (bcullerton) [inline src/main/java/uk/gov/companieshouse/accounts/user/configuration/InterceptorConfig.java:54]: Is it right that all these endpoints are hidden behind the /admin/search permission?

From account.ch it looks like there's 3 roles for api with /admin/search being used to cover the web.

/admin/roles:                         "Manage roles"
/admin/user/search:              "Search users"
/admin/user/roles:                 "Manage user roles"

See permissions.yml

File: src/main/java/uk/gov/companieshouse/accounts/user/controller/RolesController.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    public static final String PLEASE_CHECK_THE_REQUEST_AND_TRY_AGAIN = "Please check the request and try again";
+
+    public RolesController(RolesService rolesService) {
+        this.rolesService = rolesService;
+    }
+
+    @Override
+    public ResponseEntity<Void> addRole(@Valid Role role, @Pattern(regexp = "[0-9A-Za-z-_]{8,32}") final String xRequestId) {
+        
+        LOG.debug( String.format( "%s: Attempting to add the new role '%s'", xRequestId, role.getId()));
```
Comment (bcullerton) [inline src/main/java/uk/gov/companieshouse/accounts/user/controller/RolesController.java:37]: As a general rule I'd prefer trace to debug logs, it just seems to be the more standard approach for libraries so will keep logging levels consistent across other functionality. Can ignore this if debug is the standard for this app

**Review by bcullerton — CHANGES_REQUESTED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/accounts/user/configuration/InterceptorConfig.java (hunk 1 above)
Comment (dtoulcher-ch) [reply src/main/java/uk/gov/companieshouse/accounts/user/configuration/InterceptorConfig.java]: Thanks, I've updated the code to use the correct permission for the role management endpoint

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/accounts/user/controller/RolesController.java (hunk 2 above)
Comment (dtoulcher-ch) [reply src/main/java/uk/gov/companieshouse/accounts/user/controller/RolesController.java]: After discussing it has been decided to update the logs to INFO level with the exception of the getAllRoles method which has been dropped to TRACE.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by bcullerton — APPROVED** [review]

**Pull request merged**